meta-lmp-base/recipes-kernel/wireguard/wireguard-module_1.0.20200712.bb

```diff
@@ -28,6 +28,14 @@ MODULE_NAME = "wireguard"
 # with kernel-module-.
 PKG_${PN} = "kernel-module-${MODULE_NAME}"
 
+# Only set default rprovides if kernel is different than linux-lmp, assuming
+# it is older than 5.6 (version in which the module is provided by the kernel)
+python __anonymous() {
+    pn = d.getVar("PN")
+    if d.getVar("PREFERRED_PROVIDER_virtual/kernel") != "linux-lmp":
```

Comment:
Is there any way to check the kernel version of the virtual/kernel instead of linux-lmp?

Reply:
Unfortunately not from this recipe, as bitbake/OE only really fetches that level of information when executing steps on the linux-based recipe. At this point all we have is the preferred provider, and just because we have that in the machine configuration.